### DISCUSSION TITLE
Host file ignored under Safari/macOS

### DISCUSSION THREAD

**Discussion by Innoriam** [StevenBlack/hosts#2619] [Q&A]
Hello,

since I still can't find any good solution for me I'll ask you for help, maybe some of you know about the problem.

With my new MacBook came the problem, that the host file seems to not work properly. When I ping websites I'll get connected to 0.0.0.0 like I should and it fails to load the website. However in Safari it won't work anymore.

I read, that it seems to resolves websites not only through the host file but resolves it otherwise if it can't connect to it? 
I'm not so sure how it works and if it is that problem. I read something about the new http/3 being the issue.
But its very vague and there aren't many threads about it.

Does someone know a bit more and can help me? I'm looking for an easy local solution. 
I might try Pi-hole later, if I get hold of a PI, but it would be better to get a solution only for certain devices.

Thanks!

**Comment by Innoriam** [comment]
Seems like not only the host file is ignored...

I can't be the only one with that issue

**Comment by rnsavinelli** [comment]
I fixed it locally by manually adding the ipv6 loopback address to every entry inside the hosts file.
e. g. 0.0.0.0 0:0:0:0:0:0:0:1 some.url.com.ar

**Reply by Innoriam** [reply]
Thanks for your answer after so many time. The problem was not the missing of the ipv6 address. I just tried it but it still didn't work. I noticed, that in Google Chrome it worked fine. After a quick chatgpt search I found out, that the problem is private relay from iCloud. Seems like the host file is ignored if private relay is activated and only used if it is deactivated. 

So yeah, stupid path to the solution but seems I now know what it is. ChatGPT recommended a pi-hole or nextdns. Will see, if I do something like this for the future or just live with it like the last year.
